azure-pipelines.yml

```diff
@@ -0,0 +1,189 @@
+
+              [string] $newChangelogContent = $changelogContent -replace $env:currentVersion, "$env:currentVersion-dev.$(Build.BuildID)"
+
+              Write-Host " (i) Updated CHANGELOG.md content: $newChangelogContent"
+
+              $newChangelogContent | Set-Content -Path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md"
+
+        - task: CmdLine@2
+          displayName: 'Dry run publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --dry-run'
+
+        - task: CmdLine@2
+          condition: eq(variables['Build.SourceBranch'], 'refs/heads/release')
+          displayName: 'Publish'
+          inputs:
+            workingDirectory: '$(Build.SourcesDirectory)/device_calendar'
+            script: '$(flutterExecPath) pub publish --force'
+            
+  # ----------- CD Production -----------
+  - stage: Release
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    jobs: 
+     - job: Release
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: PowerShell@2
+          displayName: 'Get version from pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern $(versionNumberRegex) | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+        - task: PowerShell@2
+          displayName: 'Get version summary from CHANGELOG.md'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $summary = select-string -path "$(Build.SourcesDirectory)/device_calendar/CHANGELOG.md" -pattern "($([regex]::escape($env:currentVersion))).*" | %{ $_.Matches[0].Value }
+              
+              Write-Host "##vso[task.setvariable variable=versionSummary]$summary"
+
+        - task: GitHubReleasePublish@1
+          displayName: 'Tag a release to GitHub'
+          inputs:
+            githubEndpoint: 'GitHub Tagging'
+            manuallySetRepository: false
+            githubRepository: 'builttoroam/flutter_plugins'
+            githubTag: 'v$(currentVersion)'
+            githubReleaseTitle: '$(versionSummary)'
+            githubReleaseDraft: false
+            githubReleasePrerelease: false
+            githubIgnoreAssets: false
+            githubReleaseAsset: '$(Build.ArtifactStagingDirectory)/*'
+            githubReuseRelease: false
+            githubReuseDraftOnly: false
+            githubSkipDuplicatedAssets: false
+            githubEditRelease: false
+            githubDeleteEmptyTag: false
+
+        - task: CmdLine@2
```

Comment:
Wouldn't we want to do the dry run publish before we do the github tag?

Reply:
I think someone pointed out about this but can't remember, happy to move dry run & publish prior to github tagging

Reply:
Actually I reckon this could be a better solution as the github tag will not be created if publish fails - which means I don't need the task with the `failed()` condition to delete the github tag